### ISSUE TITLE
Data setters do not always redraw

### ISSUE THREAD

**Issue by zakandrewking** [zakandrewking/escher#279]
From Gitter

"""
Mátyás Fodor @matyasfodor Sep 05 23:31
@zakandrewking
TL;DR; how to make my changes visible if Builder. has_custom_reaction_styles is true and this._update_data(true, true, 'reaction') is not called in set_reaction_data?
Sorry, I was referencing the wrong code part. This is the change (https://github.com/zakandrewking/escher/commit/d6055d6a3ac7bbb3c278bde82571f87ea2b19cc6#diff-5772d06ee1007b34124967849a5f7231R690) in set_reaction_data that used to update the reaction colors directly. Now, _reaction_check_add_abs() returns null because I use custom reaction styles therefore the call for this._update_data(true, true, 'reaction') is prevented.
It is actually not a bad thing if I can call a series of data setter methods, and enforce update only at the end of these operations. But Builder._update_data has an underscore in its name, so I thought it's not a public method.

Zachary A. King @zakandrewking 07:54
ah, i see. yes, go ahead and call Builder._update_data. The point of the change was to minimize unnecessary redrawing, and that inevitably puts some pressure on the use to be explicit about when to redraw
although, now that i think about it, set_reaction_data is documented ... so it should probably draw every time
i'll create a new issue
"""
